device/cuda/src/seeding/seed_finding.cu

```diff
@@ -33,6 +34,15 @@
 namespace traccc::cuda {
 namespace kernels {
 
+/// CUDA kernel for running @c traccc::device::fill_prefix_sum
+__global__ void fill_prefix_sum(
+    vecmem::data::vector_view<unsigned int> sizes_view,
```

Comment:
You could switch to the proposed `traccc::device::prefix_sum_size_t` in all appropriate places in this file and in `seed_finding.sycl`. (Keeping "const correctness" in mind.)